### PR TITLE
Extend the list of possible package extensions

### PR DESCRIPTION
**Pull request by feuri**
Possible extensions are `('.pkg.tar', '.pkg.tar.gz', '.pkg.tar.bz2', '.pkg.tar.xz', '.pkg.tar.Z')`. I haven't added `'.pkg.tar.Z'` (compress) here, because it's not supported by tarfile.

### CODE REVIEW

**Comment by ushis** [comment]
hmm... not sure if there is a real need to support all those package extensions...

is makepkg supporting some of those? if yes, i'll take that. if not, i think local-repo is used for packages built by the user with makepkg, so there is no real need to support old pkg extensions... @feuri what do you think?

**Comment by feuri** [comment]
Makepkg supports all of these extensions, see `man 5 makepkg.conf`. For example [oilrush](https://aur.archlinux.org/packages.php?ID=47150) uses just tar and no compression, because else it would take really long to compress the data. (Yes, `'.tar'` is not valid, it should be `'.pkg.tar'`, but wrong package extensions should not be supported.)

**Comment by ushis** [comment]
aah, ok. i knew i saw it somewhere, but i couldnt find it. just had a look at `makepkg -h`. i'll write some comments to the diff and merge that soon.

File: localrepo/package.py [hunk 1]
```diff
@@ -157,12 +158,24 @@ def from_file(path):
...
 			raise Exception(_('File does not exist: {0}').format(path))
 
-		tmpdir = Package.get_tmpdir()
+		if tarfile.is_tarfile(path):
+			pkg = tarfile.open(path)
+
+			try:
+				pkginfo = pkg.extractfile('.PKGINFO').read().decode('utf8')
+			except:
+				raise Exception('File is not valid package: {0}'.format(path))
```
Comment (ushis) [inline localrepo/package.py:167]: missing localization here and bad exception message (i know - this was my fault). we need something like:

```
raise Exception(_('Could not read package info: {0}').format(path))
```

**Pull request merged**